optimum/onnxruntime/modeling_seq2seq.py

```diff
@@ -662,6 +758,77 @@ def __call__(self, *args, **kwargs):
         return self.forward(*args, **kwargs)
 
 
+class ORTEncoderForSpeechSeq2Seq(ORTEncoder):
+    """
+    Encoder model for ONNX Runtime inference for SpeechSeq2Seq models.
+
+    Arguments:
+        session (`onnxruntime.InferenceSession`):
+            The ONNX Runtime inference session associated to the encoder.
+    """
+
+    def prepare_io_binding(
+        self,
+        input_features: torch.FloatTensor = None,
+    ):
+        io_binding = self.session.io_binding()
+
+        # bind input ids
+        io_binding.bind_input(
+            "input_features",
+            input_features.device.type,
+            self._device.index,
+            self.name_to_np_type["input_features"],
+            tuple(input_features.shape),
+            input_features.data_ptr(),
+        )
+
+        # bind logits
+        output_shape, output_buffer = self.prepare_output_buffer(
+            batch_size=input_features.size(0),
+            sequence_length=input_features.size(2) // 2,
```

Comment:
This way to get the output shape seems to be too specific for the `Whisper` model and may not fit well for other `SpeechSeq2seq` models? Is there a way to avoid giving output shapes? @JingyaHuang 

Or if this is the only way should we rename the class to `ORTModelForWhisperConditionalGeneration`? This may lead to have different classes for each model type in future. Probably something to discuss.

Reply:
There is a way to avoid giving the output shape -> bind the output with OrtValue which will be the case for custom tasks #447 , but the flaw is that then you need to transfer ownership across frameworks which is something that we try to avoid. IMO, if you can infer the output shape, you shall bind it directly with a torch tensor.  cc. @philschmid

Reply:
Second this. We could have a general `model type -> ORTEncoder class` mapping, and `ORTModelForSpeechSeq2Seq` would use this:
```python
class ORTModelForSpeechSeq2Seq:
   ...
   ...

    def _initialize_encoder(
        self,
        session: onnxruntime.InferenceSession,
        config: transformers.PretrainedConfig,
        device: torch.device,
        use_io_binding: bool = True,
    ) -> ORTEncoderForSpeechSeq2Seq:
        return _MODEL_TYPE_TO_ORTENCODER.get(model_type, "default")(...)

Reply:
Updated